src/managers/tls.py

```diff
@@ -52,37 +52,74 @@ def set_certificate(self) -> None:
 
     def set_truststore(self) -> None:
         """Creates the unit Java Truststore and adds the unit CA."""
-        keytool_cmd = "charmed-zookeeper.keytool" if self.substrate == "vm" else "keytool"
-
         try:
-            self.workload.exec(
-                command=[
-                    keytool_cmd,
-                    "-import",
-                    "-v",
-                    "-alias",
-                    "ca",
-                    "-file",
-                    self.workload.paths.ca,
-                    "-keystore",
-                    self.workload.paths.truststore,
-                    "-storepass",
-                    self.state.unit_server.truststore_password,
-                    "-noprompt",
-                ],
-            )
+            self._import_ca_in_truststore()
+
             if self.substrate == "vm":
                 self.workload.exec(
                     command=["chown", "snap_daemon:root", self.workload.paths.truststore],
                 )
-
         except (subprocess.CalledProcessError, ops.pebble.ExecError) as e:
             if "already exists" in str(e.stdout):
+                # replace ca in four steps to prevent the truststore for being empty at any point.
+                try:
+                    if self.substrate == "vm":
+                        self.workload.exec(
+                            command=["chown", "root:root", self.workload.paths.truststore],
+                        )
```

Comment:
**question:** Why do we need to `root:root` here?

Reply:
`keytool` throws a permission error when we try to remove an entry from the truststore if we do not own the file. Maybe I am was doing something wrong, though

Reply:
Documented this in 67edf12b2844